src/nanovg_pick.c

```diff
@@ -0,0 +1,1651 @@
+#include "nanovg_internal.h"
+
+#define NVG_PICK_EPS	0.0001f
+
+// Segment flags
+enum NVGsegmentFlags {
+	NVG_PICK_CORNER	= 1,
+	NVG_PICK_BEVEL = 2,
+	NVG_PICK_INNERBEVEL	= 4,
+	NVG_PICK_CAP = 8,
+	NVG_PICK_ENDCAP = 16,
+};
+
+// Path flags
+enum NVGpathFlags {
+	NVG_PICK_HASHOLES = 1,
+	NVG_PICK_SCISSOR = 2,
+	NVG_PICK_STROKE = 4,
+	NVG_PICK_FILL = 8,
+};
+
+//#define NVG_PICK_DEBUG
+#ifdef NVG_PICK_DEBUG
+	int g_ndebugBounds = 0;
+	float g_debugBounds[256][4];
+	int g_ndebugLines = 0;
+	float g_debugLines[256][4];
+
+	#define NVG_PICK_DEBUG_NEWFRAME() \
+		{ \
+			g_ndebugBounds = 0; \
+			g_ndebugLines = 0; \
+		}
+
+	#define NVG_PICK_DEBUG_BOUNDS(bounds) memcpy(&g_debugBounds[g_ndebugBounds++][0], bounds, sizeof(float) * 4)
+	#define NVG_PICK_DEBUG_LINE(A, B) \
+		{	memcpy(&g_debugLines[g_ndebugLines][0], (A), sizeof(float) * 2); \
+			memcpy(&g_debugLines[g_ndebugLines++][2], (B), sizeof(float) * 2); }
+	#define NVG_PICK_DEBUG_VECTOR(A, D) \
+		{	memcpy(&g_debugLines[g_ndebugLines][0], (A), sizeof(float) * 2); \
+			g_debugLines[g_ndebugLines][2] = (A)[0] + (D)[0]; \
+			g_debugLines[g_ndebugLines][3] = (A)[1] + (D)[1]; \
+			++g_ndebugLines; \
+		}
+	#define NVG_PICK_DEBUG_VECTOR_SCALE(A, D, S) \
+		{	memcpy(&g_debugLines[g_ndebugLines][0], (A), sizeof(float) * 2); \
+			g_debugLines[g_ndebugLines][2] = (A)[0] + (D)[0] * (S); \
+			g_debugLines[g_ndebugLines][3] = (A)[1] + (D)[1] * (S); \
+			++g_ndebugLines; \
+		}
+#else
+	#define NVG_PICK_DEBUG_NEWFRAME() 
+	#define NVG_PICK_DEBUG_BOUNDS(bounds) 
+	#define NVG_PICK_DEBUG_LINE(A, B) 
+	#define NVG_PICK_DEBUG_VECTOR(A, D)
+	#define NVG_PICK_DEBUG_VECTOR_SCALE(A, D, S) 
+#endif									
+
+struct NVGsegment {
+	int firstPoint;				// Index into NVGpickScene::points
+	short type;					// NVG_LINETO or NVG_BEZIERTO
```

Comment:
Short chosen for size & alignment, so that no padding bytes are inserted in the struct (which I usually prefer). Changing to int will increase size and converting to char will not save any space due to alignment requirements of int & floats.